File: DoohAdManager.Model/Application/DBO/Campaign.cs
using DoohAdManager.Model.Application.INV;
using DoohAdManager.Model.Shared;

namespace DoohAdManager.Model.Application.DBO
{
    public class MvCampaign
    {
        public int Id { get; set; }
        public int TenantId { get; set; }
        public required string Name { get; set; }
        public int DurationInDays { get; set; }
        public CampaignStatus Status { get; set; }
        public string? Remarks { get; set; }
        public DateTime CreatedAt { get; set; }
        public int CreatedBy { get; set; }
        public bool IsDeleted { get; set; }
        public DateTime? DeletedAt { get; set; }
        public int? DeletedBy { get; set; }
        public List<MvCampaignDate>? Date { get; set; }
        public List<MvCampaignScreen>? Screen { get; set; }
        public List<MvCampaignMedia>? CampaignMedia { get; set; }
    }

    public class MvCampaignDate
    {
        public int Id { get; set; }
        public int CampaignId { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
    }

    public class MvCampaignScreen
    {
        public int Id { get; set; }
        public int CampaignId { get; set; }
        public int ScreenId { get; set; }
        public bool ScreenDeleted { get; set; }
        public required string ScreenName { get; set; }
    }

    public class MvCampaignCreate
    {
        public int TenantId { get; set; }
        public required string Name { get; set; }
        public string? Remarks { get; set; }
        public int CreatedBy { get; set; }
        public required List<MvCampaignDateCreate> Date { get; set; }
        public required List<MvCampaignScreenAdd> Screen { get; set; }
    }

    public class MvCampaignDateCreate
    {
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
    }

    public class MvCampaignScreenAdd
    {
        public int ScreenId { get; set; }
    }

    public class MvCampaignUpdate
    {
        public int Id { get; set; }
        public CampaignStatus Status { get; set; }
    }

    public class MvCampaignDelete
    {
        public int Id { get; set; }
        public int DeletedBy { get; set; }
    }

    public class MvCampaignFilter
    {
        public int TenantId { get; set; }
        public int? CampaignId { get; set; }
        public CampaignStatus? Status { get; set; }
        public string? Search { get; set; }
    }
}    

File: dooh-ad-manager/src/app/Application/DBO/Campaign/Models/campaign.ts
import { CampaignStatus } from '../../../../Shared/Models/enum.model';
import { MvScreenOperatingHour } from '../../../INV/Screen/Models/screen-operating-hour';
import { MvCampaignMedia } from './campaign-media';

export interface MvCampaign {
  id: number;
  tenantId: number;
  name: string;
  durationInDays: number;
  status: CampaignStatus;
  remarks: string;
  createdAt: Date;
  createdBy: number;
  isDeleted: boolean;
  deletedAt: Date;
  deletedBy: number;
  date?: MvCampaignDate[];
  screen?: MvCampaignScreen[];
  campaignMedia?: MvCampaignMedia[];
}

export interface MvCampaignDate {
  id: number;
  campaignId: number;
  startDate: Date;
  endDate: Date;
}

export interface MvCampaignScreen {
  id: number;
  campaignId: number;
  screenId: number;
  screenName: string;
  address: string;
  operatingHours: MvScreenOperatingHour[];
  screenDeleted: boolean;
}

export class MvCampaignCreate {
  tenantId: number;
  name: string;
  remarks?: string;
  createdBy: number;
  date: MvCampaignDateCreate[];
  screen: MvCampaignScreenCreate[];

  constructor() {
    this.tenantId = 1;
    this.name = '';
    this.remarks = '';
    this.createdBy = 1;
    this.date = [
      {
        startDate: null,
        endDate: null,
      },
    ];
    this.screen = [];
  }
}

export interface MvCampaignDateCreate {
  startDate: Date | null;
  endDate: Date | null;
}

export interface MvCampaignScreenCreate {
  screenId: number;
}

export interface MvCampaignUpdate {
  id: number;
  status: CampaignStatus;
}

export interface MvCampaignDelete {
  id: number;
  deletedBy: number;
}

export interface MvCampaignFilter {
  tenantId: number;
  campaignId?: number;
  status?: CampaignStatus;
  search?: string;
}
